Do not remove schemas with only a default value form allOf schemas

WellingR opened the pull request:
A schema with only a default was treated as "meta data only" this caused it to be removed from an allOf schema causing the default value to be lost.

This has been fixed in this MR

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package || exit
  ./bin/generate-samples.sh ./bin/configs/*.yaml || exit
  ./bin/utils/export_docs_generators.sh || exit
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

wing328 commented:
thanks for the PR

can you you please add a test or 2 to the model utils java test file?

wing328 commented:
cc @ross-paypay

WellingR commented:
Sorry for the delay @wing328 , but I have now added a test that reproduced the problem that I had in the inline model resolver tests.

wing328 commented:
https://github.com/OpenAPITools/openapi-generator/commit/35ba0414fbfcb514db27178e26369ff2d0fc55ed#diff-b07a3793d0d84981ddc8dc2e080ef2f9c8fa35e4a9063591981d80ab6389455eR424

i think `default` is just like `description`, `nullable` etc which cannot be handled in allOf in inline model resolver at the moment.

WellingR commented:
It is true that the inline model resolver does not do anything with this. I have a project which uses a customized generator which accesses the subtypes in allOf. Since https://github.com/OpenAPITools/openapi-generator/pull/20892 these can get lost if only a default is given. Hence this change

wing328 commented:
can you please share more about your use cases with an example schema? maybe i can come up with a solution

WellingR commented:
We have some customizations in a generator as we prefer the use of Optionals. Which is we use updated templates and have some java code to enrich some data in our customized generator.

One specific feature that we added in our custom generator is to merge a default value into a schema, which can specifically be useful for enums.

See the following snippet (which was also added in the test in this MR)

```yml
          schema:
            allOf:
              - $ref: '#/components/schemas/MyEnum'
              - default: "One"
```

Our custom generator output would pretty much add the default value to the enum definition. Perhaps that is a feature that could even be supported in the inline model resolver.
